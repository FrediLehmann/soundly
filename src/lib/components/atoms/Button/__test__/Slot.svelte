<script>
  export let Component;
</script>

<svelte:component this={Component} {...$$restProps}>Button</svelte:component>
