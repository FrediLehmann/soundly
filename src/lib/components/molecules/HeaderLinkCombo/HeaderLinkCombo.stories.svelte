<script>
  import { Meta, Story } from '@storybook/addon-svelte-csf';

  import HeaderLinkCombo from './HeaderLinkCombo.svelte';
</script>

<Meta
  title="Molecules/HeaderLinkCombo"
  component={HeaderLinkCombo}
  argTypes={{
    href: {
      control: { type: 'text' },
      table: {
        type: {
          summary: 'HREF link used for the link part of the component.'
        }
      }
    },
    headingText: {
      control: { type: 'slot' },
      table: {
        type: {
          summary: 'Named slot to pass content for the heading part.'
        }
      }
    },
    linkText: {
      control: { type: 'slot' },
      table: {
        type: {
          summary: 'Named slot to pass content for the link part.'
        }
      }
    }
  }} />

<Story name="HeaderLinkCombo">
  <HeaderLinkCombo href="#">
    <svelte:fragment slot="headingText">Heading text</svelte:fragment>
    <svelte:fragment slot="linkText">Link text</svelte:fragment>
  </HeaderLinkCombo>
</Story>
