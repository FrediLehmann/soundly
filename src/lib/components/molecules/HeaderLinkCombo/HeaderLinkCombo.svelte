<script lang="ts" context="module">
  import { Link } from '$lib/components/atoms';
</script>

<script lang="ts">
  export let href: string;
</script>

<section>
  <h1><slot name="headingText" /></h1>
  <span>/</span>
  <Link {href}><slot name="linkText" /></Link>
</section>

<style>
  section {
    display: flex;
    margin-block-end: 1.5rem;
    align-items: baseline;
  }

  h1 {
    font-size: 1.875rem;
    line-height: 2.25rem;

    margin-inline-end: 0.5rem;
  }

  span {
    margin-inline-end: 0.25rem;
  }
</style>
