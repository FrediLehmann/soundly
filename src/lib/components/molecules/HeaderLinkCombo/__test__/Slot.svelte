<script>
  export let Component;
  export let HeadingText;
  export let LinkText;
  export let href;
</script>

<svelte:component this={Component} {href}>
  <svelte:fragment slot="headingText">{HeadingText}</svelte:fragment>
  <svelte:fragment slot="linkText">{LinkText}</svelte:fragment>
</svelte:component>
