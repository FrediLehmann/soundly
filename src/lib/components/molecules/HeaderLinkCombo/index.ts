import HeaderLinkCombo from './HeaderLinkCombo.svelte';

export { HeaderLinkCombo };
