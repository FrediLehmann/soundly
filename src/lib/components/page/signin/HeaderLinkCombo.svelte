<script lang="ts" context="module">
  import { HeaderLinkCombo } from '$lib/components/molecules';
</script>

<HeaderLinkCombo href="/signup">
  <svelte:fragment slot="headingText">Sign in</svelte:fragment>
  <svelte:fragment slot="linkText">Sign up</svelte:fragment>
</HeaderLinkCombo>
