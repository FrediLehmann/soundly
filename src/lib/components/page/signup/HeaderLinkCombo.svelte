<script lang="ts" context="module">
  import { HeaderLinkCombo } from '$lib/components/molecules';
</script>

<HeaderLinkCombo href="/signin">
  <svelte:fragment slot="headingText">Sign up</svelte:fragment>
  <svelte:fragment slot="linkText">Sign in</svelte:fragment>
</HeaderLinkCombo>
