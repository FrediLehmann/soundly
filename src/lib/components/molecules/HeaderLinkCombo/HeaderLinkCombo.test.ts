/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import HeaderLinkCombo from './HeaderLinkCombo.svelte';
import Slot from './__test__/Slot.svelte';

describe('<HeaderLinkCombo /> component', () => {
  test('renders correctly', () => {
    const { component } = render(Slot, {
      props: {
        Component: HeaderLinkCombo,
        href: '/test',
        HeadingText: 'Test Heading',
        LinkText: 'Link Text'
      }
    });
    expect(component).toMatchSnapshot();
  });
});
