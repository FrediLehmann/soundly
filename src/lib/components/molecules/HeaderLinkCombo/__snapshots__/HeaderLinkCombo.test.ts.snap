// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLinkCombo /> component renders correctly 1`] = `
Slot {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      [Function],
      "Test Heading",
      "Link Text",
      "/test",
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "Component": 0,
      "HeadingText": 1,
      "LinkText": 2,
      "href": 3,
    },
    "root": <div>
      <section
        class="svelte-1z0tcs5"
      >
        <h1
          class="svelte-1z0tcs5"
        >
          Test Heading
        </h1>
         
        <span
          class="svelte-1z0tcs5"
        >
          /
        </span>
         
        <a
          class="svelte-ggsnh4"
          href="/test"
          sveltekit:prefetch="true"
        >
          Link Text
        </a>
      </section>
      
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
